main_loop_tf/validate.py

```diff
@@ -150,49 +159,58 @@ def validate(placeholders,
-                 _, summary_str) = cfg.sess.run(outs + [summary_op],
-                                                feed_dict=feed_dict)
-                cfg.sv.summary_computed(cfg.sess, summary_str,
-                                        global_step=cidx)
+                    (of_pred_batch, y_pred_batch, y_prob_batch,
+                     metrics_out, loss, _, summary_str) = cfg.sess.run(
+                         outs + [summary_op], feed_dict=feed_dict)
+                    cfg.sv.summary_computed(cfg.sess, summary_str,
+                                            global_step=cidx)
             else:
-                (y_pred_batch, y_soft_batch, mIoU, per_class_IoU, loss,
-                 _) = cfg.sess.run(outs, feed_dict=feed_dict)
+                (of_pred_batch, y_pred_batch, y_prob_batch,
+                 metrics_out, loss, _) = cfg.sess.run(
+                     outs, feed_dict=feed_dict)
+
             tot_loss += loss
 
-            # If fg/bg, just consider the foreground class
-            if len(per_class_IoU) == 2:
-                per_class_IoU = per_class_IoU[1]
+            if cfg.compute_mean_iou:
+                mIoU, per_class_IoU = metrics_out
+                # If fg/bg, just consider the foreground class
+                if len(per_class_IoU) == 2:
+                    per_class_IoU = per_class_IoU[1]
 
-            # Save the IoUs per subset (i.e., video) and their average
-            if cfg.summary_per_subset:
-                per_subset_IoUs[subset] = per_class_IoU
-                mIoU = np.mean(per_subset_IoUs.values())
+                # Save the IoUs per subset (i.e., video) and their average
+                if cfg.summary_per_subset:
+                    per_subset_IoUs[subset] = per_class_IoU
+                    mIoU = np.mean(per_subset_IoUs.values())
 
-            pbar.set_postfix({
-                'loss': '{:.3f}({:.3f})'.format(loss, tot_loss/(bidx+1)),
-                'mIoU': '{:.3f}'.format(mIoU)})
+                pbar.set_postfix({
+                    'loss': '{:.3f}({:.3f})'.format(loss, tot_loss/(bidx+1)),
+                    'mIoU': '{:.3f}'.format(mIoU)})
         else:
             if cidx % cfg.val_summary_freq == 0:
-                y_pred_batch, y_soft_batch, summary_str = cfg.sess.run(
-                    outs[:2] + [summary_op], feed_dict=feed_dict)
-                mIoU = 0
-                cfg.sv.summary_computed(cfg.sess, summary_str,
+                fetch_dict = cfg.sess.run(outs_summary, feed_dict=feed_dict)
+
+                cfg.sv.summary_computed(cfg.sess,
+                                        fetch_dict['summary_op'],
                                         global_step=cidx)
             else:
-                y_pred_batch, y_soft_batch = cfg.sess.run(outs[:2],
-                                                          feed_dict=feed_dict)
+                fetch_dict = cfg.sess.run(outs, feed_dict=feed_dict)
+                mIoU = 0
```

Comment:
I believe mIoU should be set to 0 also in the other branch (i.e., once for both branches, before L193)